pkg/crypto/ed25519.go

```diff
@@ -0,0 +1,101 @@
+package crypto
+
+import (
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"io/ioutil"
+	"os"
+
+	"golang.org/x/crypto/ed25519"
+)
+
+var (
+	// PubKeyIdentifier is the PEM public key identifier
+	PubKeyIdentifier = "PUBLIC KEY"
+	// PrivKeyIdentifier is the PEM private key identifier
+	PrivKeyIdentifier = "PRIVATE KEY"
+	// PEMCipher is the PEM encryption algorithm
+	PEMCipher = x509.PEMCipherAES256
+	// PubKeyFilePermissions are the public key file perms
+	PubKeyFilePermissions os.FileMode = 0644
+	// PrivKeyFilePermissions are the private key file perms
+	PrivKeyFilePermissions os.FileMode = 0600
+)
+
+// LoadPublicKeyFromFile loads PEM formatted ED25519 public key from file.
+func LoadPublicKeyFromFile(publicKeyPath string) ([]byte, error) {
+	x509PEM, err := ioutil.ReadFile(publicKeyPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Parse x509 PEM file
+	block, _ := pem.Decode(x509PEM)
+	if block == nil || block.Type != PubKeyIdentifier {
+		return nil, errors.New("Can't decode PEM file")
+	}
+
+	return block.Bytes, nil
+}
+
+// LoadPrivateKeyFromFile loads PEM formatted ED25519 private key from file.
+func LoadPrivateKeyFromFile(privateKeyPath string, password []byte) ([]byte, error) {
+	x509PEM, err := ioutil.ReadFile(privateKeyPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Parse x509 PEM file
+	block, _ := pem.Decode(x509PEM)
+	if block == nil || block.Type != PrivKeyIdentifier {
+		return nil, errors.New("Can't decode PEM file")
+	}
+
+	// Check for encrypted PEM format
+	if x509.IsEncryptedPEMBlock(block) {
+		decryptedKey, err := x509.DecryptPEMBlock(block, password)
+		if err != nil {
+			return nil, err
+		}
+		return decryptedKey, nil
+	}
+
+	return block.Bytes, nil
+}
+
+// GeneratED25519Key generates a ED25519 keypair
+func GeneratED25519Key(password []byte, privateKeyFilePath string, publicKeyFilePath string) error {
+	pubKey, privKey, err := ed25519.GenerateKey(rand.Reader)
+	if err != nil {
+		return err
+	}
+
+	var privBlock = &pem.Block{
+		Type:  "PRIVATE KEY",
+		Bytes: privKey,
+	}
+
+	var pubBlock = &pem.Block{
+		Type:  "PUBLIC KEY",
+		Bytes: pubKey,
+	}
+
+	var privateKey []byte
+	if password != nil || len(password) < 1 {
```

Comment:
you can simply check that `len(password) < 1`, it will also work when `password == nil` (as long as `password`'s type remains `[]byte`)

Reply:
okay